services/registration/src/routes/grading.ts

```diff
@@ -215,35 +224,46 @@ gradingRouter.route("/actions/submit-review").post(
       throw new BadRequestError("One or more of the method body arguments is missing.");
     }
 
-    const numCalibrationScoresForGroup = grader.calibrationScores.filter(
-      val => val.group === req.body.gradingGroup
-    ).length;
+    const gradingGroup = req.body.gradingGroup as GradingGroupType;
+
+    let criteriaScores = grader.calibrationScores.find(
+      calibrationScore => calibrationScore.group === gradingGroup
+    )?.criteriaScores;
+
+    const numCalibrationScoresForGroup = criteriaScores ? criteriaScores.length : 0;
 
     // Checks to make sure the user has completed all the calibration questions and that this request
     // is for a calibration question
     if (
-      numCalibrationScoresForGroup < calibrationQuestionMapping[req.body.gradingGroup].length &&
+      numCalibrationScoresForGroup < calibrationQuestionMapping[gradingGroup].length &&
       (!req.body.isCalibrationQuestion || req.body.applicationId)
     ) {
       throw new BadRequestError(
         "Cannot submit a score for this group yet. Calibration questions are not complete."
       );
     }
 
-    if (numCalibrationScoresForGroup < calibrationQuestionMapping[req.body.gradingGroup].length) {
-      grader.calibrationScores.push({
-        group: req.body.gradingGroup,
+    if (numCalibrationScoresForGroup < calibrationQuestionMapping[gradingGroup].length) {
```

Comment:
```suggestion
    if (numCalibrationScoresForGroup < calibrationQuestionMapping[gradingGroup].length) {
      if (!criteriaScores) {
        criteriaScores= [{
          criteria: req.body.criteria,
          score: req.body.score,
        }];
        grader.calibrationScores.push({
          group: gradingGroup,
          criteriaScores,
        });
      }
```

Reply:
`criteriaScores` is an array so it's reference is stored in `calibrationScores` and gets updated on the later push anyway so I think this is unneeded